core/src/net/udp_socket.cpp

```diff
@@ -0,0 +1,186 @@
+
+    CUBOS_INFO("Bind UDP socket '{}' at address {} with port '{}'", mSock, address.toString(), port);
+    return true;
+}
+
+void UdpSocket::unbind() const
+{
+    if (mSock != INVALID_SOCKET)
+    {
+        CUBOS_WARN("Closing UDP socket: '{}'", mSock);
+#ifdef _WIN32
+        closesocket(mSock);
+        WSACleanup();
+#else
+        close(mSock);
+#endif
+    }
+}
+
+bool UdpSocket::send(const void* data, std::size_t size, const Address& address, uint16_t port) const
+{
+    if (mSock == INVALID_SOCKET)
+    {
+        CUBOS_ERROR("Invalid socket: {}", error());
+        return false;
+    }
+
+    sockaddr_in addr;
+    addr.sin_family = AF_INET;
+    addr.sin_port = htons(port);
+    addr.sin_addr.s_addr = inet_addr(address.toString().c_str());
+
+#ifdef _WIN32
+    auto res = sendto(mSock, static_cast<const char*>(data), static_cast<int>(size), 0,
+                      reinterpret_cast<struct sockaddr*>(&addr), sizeof(addr));
+#else
+    auto res = sendto(mSock, data, size, 0, reinterpret_cast<struct sockaddr*>(&addr), sizeof(addr));
+#endif
+    if (res == INVALID_SOCKET)
+    {
+        CUBOS_ERROR("Failed to send UDP data: {}", error());
+        return false;
+    }
+
+    CUBOS_INFO("Sent UDP message to socket '{}' at address {} and port '{}'", mSock, address.toString(), port);
+    return true;
+}
+
+bool UdpSocket::receive(void* data, std::size_t size, std::size_t& received, Address& address) const
+{
+    if (mSock == INVALID_SOCKET)
+    {
+        CUBOS_ERROR("Invalid socket: {}", error());
+        return false;
+    }
+
+    sockaddr_in addr;
+    socklen_t addrLen = sizeof(addr);
+
+#ifdef _WIN32
+    auto len = recvfrom(mSock, static_cast<char*>(data), static_cast<int>(size), 0,
+                        reinterpret_cast<struct sockaddr*>(&addr), &addrLen);
+#else
+    auto len = recvfrom(mSock, data, size, 0, reinterpret_cast<struct sockaddr*>(&addr), &addrLen);
+#endif
+
+    if (len == INVALID_SOCKET)
+    {
+        CUBOS_ERROR("Failed to receive UDP data: {}", error());
+        return false;
+    }
+
+    received = static_cast<std::size_t>(len);
+    address = Address::from(std::string(inet_ntoa(addr.sin_addr))).value();
+
+    CUBOS_INFO("Received UDP message at socket '{}' from address '{}'", mSock, address.toString());
+    return true;
+}
+
+bool UdpSocket::setBlocking(bool blocking)
+{
+    bool success = false;
+
+#ifdef _WIN32
+    u_long flags = blocking ? 0 : 1;
+    success = NO_ERROR == ioctlsocket(mSock, FIONBIO, &flags);
+#else
+    mBlocking = blocking;
+
+    int flags = fcntl(mSock, F_GETFL, 0);
+
+    if (blocking)
+    {
+        flags &= ~O_NONBLOCK;
+    }
+    else
+    {
+        flags |= O_NONBLOCK;
+    }
+
+    success = fcntl(mSock, F_SETFL, flags) == 0;
+#endif
+    if (!success)
+    {
+        CUBOS_ERROR("Failed to change block flag of socket '{}'", mSock);
+    }
+    else
+    {
+        CUBOS_INFO("UDP socket {} is now '{}'", mSock, blocking ? "blocking" : "non-blocking");
+    }
```

Comment:
This shouldn't be necessary, as this is part of the core functionality and users should just expect it to work.